src/format/parse.rs

```diff
@@ -574,6 +574,41 @@ mod tests {
                 assert_eq!(parse_all($fmt, &$items), Ok(expected))
             });
         }
+        macro_rules! lit {
```

Comment:
Having two copies of this does not seem like an improvement.

Reply:
They were the reason the submodule declaration where low in the file: only macros declared on the line above are available to submodules.

> Having two copies of this does not seem like an improvement.

I kind of agree. They were used only in one method `StrftimeItems` until a large test in `format::parse` was added.
Because we duplicate macro's in many places to make tests shorter it seemed acceptable to me.

Reply:
I don't mind changing it if you see a better way.

Reply:
I think my solution would be to make these normal functions (maybe with `#[inline(always)]` instead of macros) and/or import enum variants into local namespaces instead. IMO none of these really need to be macros.